### PR TITLE
test: migrate modals to vitest

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Migrate modal related tests to vitest

# Changes

- migrate test
- add `<svelte:options accessors/>` to `TransactionModal.svelte` to expose functions
- remove test "should disable next button if network not selected" in `TransactionModal.spec` **

# TODO

Few modals in canisters, sns and neurons were not migrated.

# Notes

** I don't know why this test was passing before as the dropdown is automatically selected when an account is entenred. So there is not good way to test if the "next" button is disabled because the dropdown can be empty only if the account entered is invalid and if invalid, the "next" button is disabled anyway.

### CODE REVIEW

**Comment by peterpeterparker** [comment]
@lmuntaner there are few tests I left beside to provide the PR. Can you take care of those?

**Comment by lmuntaner** [comment]
> @lmuntaner there are few tests I left beside to provide the PR. Can you take care of those?

Yes, I can. I want to finish the Tag and then I can take care of the rests of the test.

File: frontend/src/vitests/lib/modals/common/ConfirmationModal.spec.ts [hunk 1]
```diff
@@ -22,15 +18,17 @@ describe("ConfirmationModal", () => {
     expect(getByText(noButtonText)).toBeInTheDocument();
   });
 
-  it("should trigger nnsClose", (done) => {
-    const { getByText, component } = render(ConfirmationModal);
-    component.$on("nnsClose", () => done());
-    fireEvent.click(getByText(noButtonText));
-  });
+  it("should trigger nnsClose", () =>
+    new Promise<void>((done) => {
```
Comment (lmuntaner) [inline frontend/src/vitests/lib/modals/common/ConfirmationModal.spec.ts:22]: So, if the function returns a promise, vitest will wait for the promise to resolve?

Did you see it failing?

File: frontend/src/vitests/lib/modals/sns/SaleInProgressModal.spec.ts [hunk 2]
```diff
@@ -19,6 +15,8 @@ describe("SaleInProgressModal", () => {
   it("should render progress in a modal", async () => {
     const { getByTestId } = await renderSaleInProgressModal();
 
-    await waitFor(expect(getByTestId("in-progress-warning")).not.toBeNull);
+    await waitFor(() =>
+      expect(getByTestId("in-progress-warning")).not.toBeNull()
```
Comment (lmuntaner) [inline frontend/src/vitests/lib/modals/sns/SaleInProgressModal.spec.ts:19]: Thanks!

**Review by lmuntaner — APPROVED** [review]
Thanks!

Just a minor comment.

File: frontend/src/vitests/lib/modals/common/ConfirmationModal.spec.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/vitests/lib/modals/common/ConfirmationModal.spec.ts:22]: This is just the official way to migrate `done()` in vitest applied.
https://vitest.dev/guide/migration.html#done-callback

Comment (lmuntaner) [reply frontend/src/vitests/lib/modals/common/ConfirmationModal.spec.ts:22]: thanks for the explanation!

**Pull request merged**